### PR TITLE
Remove `context` and `should` from a bunch of tests

### PR DESCRIPTION
**Pull request by josemigallas**
### Remove `shoulda-context` (part 1)

First step is removing usage of `context` and `should` in favor of organizing in classes and using `test`.

#### Affected files

* test/unit/array_test.rb
* test/unit/abilities/forums_test.rb
* test/unit/month_test.rb
* test/unit/referrer_filters_test.rb
* test/unit/cinstance/trial_test.rb

#### JIRA
[THREESCALE-7907: Remove shoulda-context > THREESCALE-7939: Remove all contexts](https://issues.redhat.com/browse/THREESCALE-7939)

### CODE REVIEW

File: test/unit/abilities/forums_test.rb [hunk 1]
```diff
@@ -1,312 +1,239 @@
...
       @provider = FactoryBot.create(:provider_account)
+      @admin_user = @provider.admins.first
+      # admin = FactoryBot.create(:admin, account: @provider)
+      @provider_user = FactoryBot.create(:user, account: @provider)
+
       @forum = @provider.forum
+      @topic = FactoryBot.create(:topic, forum: @forum)
+
+      @buyer = FactoryBot.create(:buyer_account, provider_account: @provider)
+      @buyer_user = FactoryBot.create(:user, account: @buyer)
```
Comment (akostadinov) [inline test/unit/abilities/forums_test.rb:17]: This seems to be bloating setup for all tests making them slower, maybe not a big deal. Not blocker definitely.

**Review by akostadinov — APPROVED** [review]

File: test/unit/abilities/forums_test.rb (hunk 1 above)
Comment (josemigallas) [reply test/unit/abilities/forums_test.rb]: OK I refactored it a bit more and also increased coverage by creating 2 branches: public and private forum.

File: test/unit/abilities/forums_test.rb [hunk 2]
```diff
@@ -6,51 +6,72 @@ module Abilities
...
-      @admin_user = @provider.admins.first
-      # admin = FactoryBot.create(:admin, account: @provider)
-      @provider_user = FactoryBot.create(:user, account: @provider)
+    end
 
-      @forum = @provider.forum
-      @topic = FactoryBot.create(:topic, forum: @forum)
+    class PublicForumTest < ForumsTest
+      def setup
+        super
```
Comment (akostadinov) [inline test/unit/abilities/forums_test.rb:13]: was not inner class already invoke outher class `#setup`? I'm not sure, just asking whether `@provider = FactoryBot.create(:provider_account)` would be called twice.

**Review by akostadinov — DISMISSED** [review]
Looks even nicer!

pls check my question, feel free to merge in case it is not an issue.

File: test/unit/abilities/forums_test.rb (hunk 2 above)
Comment (josemigallas) [reply test/unit/abilities/forums_test.rb:13]: Not sure if I understand the question but I'm positive provider is not instantiated twice

**Pull request merged**